indico/web/client/js/react/components/EditableSubmissionButton.jsx

```diff
@@ -0,0 +1,62 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React, {useCallback, useState} from 'react';
+import PropTypes from 'prop-types';
+import {Button, Modal} from 'semantic-ui-react';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+
+import FileManager from '../../../../../modules/events/editing/client/js/components/FileManager';
+
+export default function EditableSubmissionButton({uploadURL, fileTypes, type, submitRevisionURL}) {
+  const [submissionFiles, setSubmissionFiles] = useState({});
+  async function submitRevision() {
+    try {
+      await indicoAxios.put(submitRevisionURL, {
+        files: submissionFiles,
+      });
+    } catch (e) {
+      handleAxiosError(e);
+    }
+  }
+
+  return (
+    <Modal
+      trigger={
+        <button type="submit" className="i-button highlight">
+          Submit Files
+        </button>
+      }
+      closeIcon
+    >
+      <Modal.Header>Submit a {type}</Modal.Header>
+      <Modal.Content>
+        <FileManager
+          onChange={useCallback(fts => {
```

Comment:
having useCallback inlined here doesn't look great. `const handleChange = useCallback(...);` and then just pass `onChange={handleChange}` here